### ISSUE TITLE
[BUG] Can't select long entries in the "extra content" field

### ISSUE THREAD

**Issue by InconsolableCellist** [zeapo/Android-Password-Store#2362]
### Describe the bug

I enter passwords in my files that Password Store interprets as being in the "extra content" field. (I.e., the first entry in the file isn't necessarily the password.) Thus I want to select some text from the extra content field and copy it. However, entries long enough to scroll the window make selecting text impossible.

## Steps to Reproduce: 
1. Have a password entry that has a lot of extra content (additional login information, aliases, old passwords, etc.)
2. Open Password Store in portrait mode on your phone
3. Locate the text you want to copy from the extra content field; tap and hold to initiate a normal selection

## Observed Results
Password Store scrolls the window halfway off-screen and selects a big block of text. The grippers cannot be moved up far enough to modify the selection, as the window won't scroll and the header at the top of the app obscures text.

This makes selecting passwords for some of my logins impossible.

## Expected Results
1. Text selection doesn't scroll the window
2. Text selection doesn't select a big chunk of text by default (probably a result of the window scrolling halfway down on its own)
3. The selection grippers can be dragged around and the window will scroll if necessary

## Environment: 
Samsung Z Fold 4 
Password Store 1.13.5

**Comment by InconsolableCellist** [comment]
Example password store file to try:

```
email server login:
username123@email.com
oldpassword1
oldpassword2
currentpassword

information I gave for this site:
John Doe
123 State St.
Springfield, IL 01234 
(555) 123-456-7890

website provides email aliases: 
username123@email.*

"*" can be any of the following:
.com
.net
.org
.de
.icu
.uk
.ru
.info
.top
.xyz
.tk
.cn
.ga
.cf
.nl

Contacted support on 1/2/23
Ticket number: 123456
Callback number: +1 (555) 555-5555
Support ticket number: #123456
```

Try selecting `currentpassword` in this text, assuming it's long enough to significantly go off-screen on the device you're  testing on.

**Comment by msfjarvis** [comment]
I can look into this at some point but I don't quite understand why can you not just make the password be the very first line? All `pass` compatible CLIs and GUIs special case the first line as a password field, and you can use a blank line for entries where a password does not exist.

**Comment by InconsolableCellist** [comment]
Thanks for the quick reply. I don't understand how the inability to select extra content makes this low priority. Extra content isn't just a decoration, it contains useful information you're often going to want to enter into a site. Other clients (like the Linux one) don't have this issue.

The workaround you suggest would still mean that extra content can't be reliably selected when it's large enough to trigger scrolling. Other Android apps can don't have this problem with long text fields.

**Comment by msfjarvis** [comment]
If you disagree with the prioritization, feel free to fix the issue for me. There's an upper limit on how much time I can give to this project that has never made me a dime, and I'd like to retain my agency on how I choose to spend it.

When I asked why you chose to put the password in a non-standard location, it was not a workaround, it was a question. There are direct benefits to following the standard set by `pass`, and I simply wanted to know if there was any reason why you were doing things the way you were.

**Comment by InconsolableCellist** [comment]
I want to make sure you understand the bug as reported, in that extra content isn't a useless field. Low priority to me could indicate that you don't see much value in being able to select text from the extra content field.

>When I asked why you chose to put the password in a non-standard location, it was not a workaround, it was a question.

Typically I have to change my passwords over time, and the first entry gets out of date or is information about the website.

**Comment by msfjarvis** [comment]
> I want to make sure you understand the bug as reported, in that extra content isn't a useless field. Low priority to me could indicate that you don't see much value in being able to select text from the extra content field.

I am very aware that extra content is not a useless field, and the bug you've encountered is precisely because of that fact. The vast majority of users use key-value pairs separated by colons to store data in their extra content section, which is why we built out a parser that shows them as individual entries in the screen rather than a bulk mass of text. The selection doesn't work properly because the entire screen is scrollable and so is the text field containing your very long unstructured extra content, and nested scrolling in parallel directions needs special care to get right. We would've worked around it if we ever encountered it before, just as we will now.

**Comment by Skrilltrax** [comment]
@InconsolableCellist Have you tried using the snapshot builds? I tested this and can reproduce the bug on 1.13.5 but it is fixed in the latest snapshot.

**Comment by msfjarvis** [comment]
Closed for inactivity and https://github.com/android-password-store/Android-Password-Store/issues/2362#issuecomment-1513789894
